src/Mod/Path/PathScripts/PathProfile.py

```diff
@@ -56,8 +56,8 @@
 
 
 # Qt translation handling
-def translate(context, text, disambig=None):
-    return QtCore.QCoreApplication.translate(context, text, disambig)
+def tr(text):
+    return QtCore.QT_TRANSLATE_NOOP("PathProfile", text)
```

Comment:
I don't know if this is correct. Wouldn't this just pass through the text untranslated?

Reply:
Looked into it further, this will not play well with `lupdate`. it does static analysis only, so the function HAS to be called QT_TRANSLATE_NOOP and include the context as an argument, otherwise it will not get extracted with the correct context. the good news is that you dont have to fully qualify it, so you could just bring it into the namespace at the top of the file like so:

```python
from PySide.QtCore import QT_TRANSLATE_NOOP
```

Another shorthand possibility is to use that function once at the top of the file to assign the string to a variable and then use that variable in the body of the code, specially if the string is repeated many times.

I am still not sure how the GUI knows which context to use, but it evidently works on other workbenches.

Reply:
Another shorthand option is if the build system is set up to add an alias when running lupdate so the function can be called with a shorter name like `-tr-function-alias QT_TRANSLATE_NOOP+=_TN` and then in the code
```python
from PySide.QtCore import QT_TRANSLATE_NOOP as _TN
```
or somesuch

Reply:
@iromero91 is correct -- `lupdate` is basically just a simple text parser. Also, I thought our discussion was that you needed to use `QtCore.QCoreApplication.translate()` here: you are trying to get the actual translated string here, right? Not just mark the text for translation with a later call to `translate()`?

Reply:
Nevermind that, the python version of lupdate does not seem to have aliases in qt5. In any case i confirmed that these changes make all the translations end up in the `@Default` context

Reply:
@chennes actually I think we need the nop here as the translate call is done by the freecad gui.

Reply:
Not the way it's being used here. First, that won't apply to any of the console output (though I was under the impression we didn't translate console output). Second, take the following code:
```
             "JoinType": [
                 (tr("Round"), "Round"),
                 (tr("Square"), "Square"),
                 (tr("Miter"), "Miter"),
             ],
```
It's critical in this use that the first string of the pair be the translated string in the internal code (not just as displayed in the UI). So this cannot use the NOOP, it must actually query the translation system.

Reply:
I'm getting more confused by the minute.  The lupdate stuff makes sense, so I'll unwrap those calls.  
Everything else has my head in a twist.   

Realthunder said (in Telegram)  "I notice in existing Path code, some Path object pass translate()'ed string as documentation for properties of newly created object, this means, the those translated string are saved into file, another user in a different language will not be able to read"   Example: PathToolBit.py:140

This makes sense but the alternative is that the documentation is stored in english and untranslated so another user in a different language still can't read them.  


I grepped through everything in /MOD looking for how other workbenches are doing this.  There are a million calls to QT_TRANSLATE_NOOP  and only TEN calls to QCoreApplication.translate and all of those are in two files in the Draft wb.   Is Path the only workbench that's actually trying to do translation in the GUI?

> (though I was under the impression we didn't translate console output). 

Are there written standards for how translation should be handled?

Reply:
If you're passing the string into something that's going to get `translate()` called on it later, you use NOOP so that the string gets extracted by `lupdate`, but the untranslated string gets passed around. This is common when dealing with Command, for example. Command itself calls `translate()` as it loops over all the commands puts them into the UI, but since it's working with variables rather than string literals, you have to make sure there is a `QT_TRANSLATE_NOOP` someplace with the literal in it so that the string exists in the translation database when `translate()` goes to find it. Property is similar: your docstrings there should use NOOP because the Property classes call translate() later on when actually showing those strings to the user. Since Command and Property make up like 95% of FreeCAD, that's why you see such prevalence of the NOOP.

In addition, for most of the UI there are functions associated with each dialog called `retranslate()` or `retranslateUi()` (or something along those lines) -- those functions call `translate()` on each widget (though often with a string literal, so it actually doesn't matter what strings you use elsewhere in the code, they just get overwritten!).

In your case you're doing things a bit differently: before this PR you were relying on the text content of the widget to function as an enum. Of course, as soon as translation happens that all breaks. So instead you are now attaching an extra, non-translated bit of data to each widget, so you don't care about the text that's displayed. You can either do the translation yourself when you create the widget, or you can have a `retranslate()` function that handles it later. As long as the user data doesn't get translated, it doesn't matter which approach you take.